### PR TITLE
build: support static builds for Swift

### PR DESCRIPTION
**Pull request by compnerd**
Adjust the paths for static runtime builds of Swift.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @kevints

**Comment by kevints** [comment]
@compnerd looking at the build the test programs are failing due to undefined symbols in ICU. This is due to the once-private link dependencies of swiftCore being needed (as there's no portable way for unix archives to contain library dependencies). The dispatch buildsystem currently creates an IMPORTED target for the swiftCore library without regard to its dependencies. This happens to work in the shared case but doesn't in the static case, but to make it work robustly is equivalent to duplicating the `-static-stdlib` feature of `swiftc` in the dispatch buildsystem, which is the feature we're trying to fix. A more robust approach would be to use swiftc (with the -static-stdlib flag when BUILD_SHARED_LIBS is NO) to link these test programs. We could also use an exported swiftCore target from the swift build system but I suspect that's not currently tractable.

**Comment by gottesmm** [comment]
@kevints We support creating exported targets from the parent swift cmake.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift/pull/19834

@swift-ci please test

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,38 +44,32 @@ dispatch_common_warnings()
 option(ENABLE_DISPATCH_INIT_CONSTRUCTOR "enable libdispatch_init as a constructor" ON)
 set(USE_LIBDISPATCH_INIT_CONSTRUCTOR ${ENABLE_DISPATCH_INIT_CONSTRUCTOR})
 
+# NOTE(abdulras) this is the CMake supported way to control whether we generate
```
Comment (kevints) [inline CMakeLists.txt]: Consider linking straight to the reference: https://cmake.org/cmake/help/v3.4/variable/BUILD_SHARED_LIBS.html

Comment (compnerd) [reply CMakeLists.txt]: I do like that better; I think Im going to do that in a follow up (to get this merged first).  This was just moving the option around.

**Comment by compnerd** [comment]
Also note that this fixes the builds for me with tests after @DougGregor's change (layout differences between Darwin and Linux bites us again).  My follow up change will actually use the swift compiler to link the Swift SDK overlay and split it out, so that actually makes all this significantly more robust and easier to work with.

**Comment by gottesmm** [comment]
This LGTM from the swift side of things. I think the cmake will work fine as well given that we are essentially going through cmake config and performing an identity transformation (since swift's config provides that same information). @MadCoder hows this look to you?

**Comment by compnerd** [comment]
@MadCoder - ping?

**Pull request merged**